### ISSUE TITLE
`master` build fails with `enable-buildtest-c++ --strict-warnings`

### ISSUE THREAD

**Issue by romen** [openssl/openssl#11942]
Travis failure log: https://travis-ci.org/github/openssl/openssl/jobs/690719793

```
test/ossl_shim/ossl_shim.cc: In function 'int bssl::TicketKeyCallback(SSL*, uint8_t*, uint8_t*, EVP_CIPHER_CTX*, HMAC_CTX*, int)':
test/ossl_shim/ossl_shim.cc:394:73: warning: 'int HMAC_Init_ex(HMAC_CTX*, const void*, int, const EVP_MD*, ENGINE*)' is deprecated [-Wdeprecated-declarations]
   if (!HMAC_Init_ex(hmac_ctx, kZeros, sizeof(kZeros), EVP_sha256(), NULL) ||
                                                                         ^
In file included from include/openssl/opensslv.h:109:0,
                 from include/openssl/macros.h:11,
                 from include/openssl/opensslconf.h:14,
                 from include/openssl/macros.h:10,
                 from include/openssl/bio.h:14,
                 from test/ossl_shim/include/openssl/base.h:14,
                 from test/ossl_shim/packeted_bio.h:13,
                 from test/ossl_shim/ossl_shim.cc:14:
include/openssl/hmac.h:39:22: note: declared here
 DEPRECATEDIN_3_0(int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
                      ^
include/openssl/macros.h:38:39: note: in definition of macro 'DECLARE_DEPRECATED'
 #     define DECLARE_DEPRECATED(f)    f __attribute__ ((deprecated));
                                       ^
include/openssl/hmac.h:39:1: note: in expansion of macro 'DEPRECATEDIN_3_0'
 DEPRECATEDIN_3_0(int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
 ^~~~~~~~~~~~~~~~
```

After dbde4726889a19af0a718fe9c5542f39c81acbd3 `test/ossl_shim/ossl_shim.cc` needs to be update to include

```
/*
 * HMAC low level APIs are deprecated for public use, but still ok for internal
 * use.
 */
#include "internal/deprecated.h"
```

or even better replacing https://github.com/openssl/openssl/blob/057452a06f02eb24d8008b975cde32d51361a2cf/test/ossl_shim/ossl_shim.cc#L394 with the newer way to access HMAC.

**Comment by romen** [comment]
@paulidale can you have a look at this?

**Comment by paulidale** [comment]
See #11944

**Comment by romen** [comment]
Fixed via #11944 :

- 1bdd86fb1c ossl_shim: add deprecation guards around the -use-ticket-callback option.

Thanks @paulidale !
